src/redisgears_python.c

```diff
@@ -2365,7 +2449,59 @@ static PyObject* scriptRunnerAddInput(PyObject *cls, PyObject *args){
         PyErr_SetString(GearsError, "Given argument is not of type PyTensor");
         return NULL;
     }
-    RedisAI_ScriptRunCtxAddInput(pys->s, pyt->t);
+    RAI_Error* err;
+    RedisAI_InitError(&err);
+    RedisAI_ScriptRunCtxAddInput(pys->s, pyt->t, err);
+    if (RedisAI_GetErrorCode(err) != RedisAI_ErrorCode_OK) {
+        PyErr_SetString(GearsError, RedisAI_GetError(err));
+        RedisAI_FreeError(err);
+        return NULL;
+    }
+    RedisAI_FreeError(err);
+    return PyLong_FromLong(1);
+}
+
+static PyObject* scriptRunnerAddInputList(PyObject *cls, PyObject *args){
+    verifyRedisAILoaded();
+    if(PyTuple_Size(args) != 2){
+        PyErr_SetString(GearsError, "Wrong number of arguments given to scriptRunnerAddInput");
+        return NULL;
+    }
+    PyTorchScriptRunner* pys = (PyTorchScriptRunner*)PyTuple_GetItem(args, 0);
+    if(!PyObject_IsInstance((PyObject*)pys, (PyObject*)&PyTorchScriptRunnerType)){
+        PyErr_SetString(GearsError, "Given argument is not of type PyTorchScriptRunner");
+        return NULL;
+    }
+
+    PyObject* py_tensorList = (PyListObject*)PyTuple_GetItem(args, 1);
+    if(!PyObject_IsInstance(py_tensorList, (PyObject*)&PyList_Type)){
+        PyErr_SetString(GearsError, "Given argument is not of type PyList");
+        return NULL;
+    }
+
+    size_t len = PyList_Size(py_tensorList);
+    RAI_Tensor** tensorList = array_new(RAI_Tensor*, len);
+    for(size_t i=0; i < len; i++){
+        PyTensor* pyt = (PyTensor*)PyList_GetItem(py_tensorList, i);
+        if(!PyObject_IsInstance((PyObject*)pyt, (PyObject*)&PyTensorType)){
+            PyErr_SetString(GearsError, "Given argument is not of type PyTensor");
+            array_free(tensorList);
+            return NULL;
+        }
+        tensorList = array_append(tensorList, pyt->t);
+    }
+
+    RAI_Error* err;
+    RedisAI_InitError(&err);
+    RedisAI_ScriptRunCtxAddInputList(pys->s, tensorList, array_len(tensorList), err);
```

Comment:
We need to consider who is taking and giving ownership here, does RedisAI get ShallowCopy of all the tensors in the list?

Reply:
redisAI generates a shallow copy

Reply:
Well rethinking about it now, the ref count is not increased atomically so theoretically we have an issue here and we need to take the GIL before call this function (same for scriptRunnerAddInput).